// midgard-core/lib/main.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _redux = require("redux");

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var Core = function Core() {
  var _this = this;

  _classCallCheck(this, Core);

  this.configureStore = function (reducers, preloadedState, enhancer) {
    _this.store = (0, _redux.createStore)(reducers, preloadedState, enhancer);
    return _this.store;
  };

  this.combineReducers = function (reducers) {
    return (0, _redux.combineReducers)(reducers);
  };

  this.applyMiddleware = function (middlerware) {
    return (0, _redux.applyMiddleware)(middlerware);
  };

  this.getState = function () {
    return _this.store && _this.store.state;
  };

  this.getDispatch = function () {
    if (_this.store) {
      return _this.store.dispatch;
    }
  };
};

var core = new Core();
var _default = core;
exports.default = _default;
module.exports = exports["default"];
